source_code/main_mcu/src/LOGIC/logic_gui.c

```diff
@@ -257,11 +257,41 @@ void logic_gui_erase_user(void)
     logic_smartcard_handle_removed();
 }
 
-/*! \fn     logic_gui_display_login_password(child_cred_node_t* child_node)
-*   \brief  Display login and password on the screen
+/*! \fn     logic_gui_create_TOTP_display_str(cust_char_t *TOTP_str, uint8_t TOTP_str_len, uint8_t remaining_secs)
+*   \brief  Create the TOTP display string: "<TOTP> [remaining secs]", example: "123456 [30]"
+*   \param  TOTP_str       String containing the TOTP
+*   \paran  TOTP_str_len   Total length of the TOTP_str
+*   \param  numDigits      Number of digits in TOTP
+*   \param  remaining_secs Number of seconds remaining
+*/
+static void logic_gui_create_TOTP_display_str(cust_char_t *TOTP_str, uint8_t TOTP_str_len, uint8_t numDigits, uint8_t remaining_secs)
+{
+    /*
+    * Ensure we have enough room for TOTP + remaining secs
+    * Number of TOTP digits + space + '[' + 2-digit remaining secs + ']'
+    */
+    uint8_t total_length = numDigits + 1 + 1 + 2 + 1;
+
+    if (remaining_secs < LOGIC_GUI_TOTP_MAX_TIMESTEP && TOTP_str_len > total_length)
+    {
+        uint8_t curr_index = numDigits;
+
+        TOTP_str[curr_index++] = ' ';
+        TOTP_str[curr_index++] = '[';
+
+        utils_itoa(remaining_secs, 2, &TOTP_str[curr_index], TOTP_str_len - curr_index);
```

Comment:
can you hardcode "2" for the str_len parameter here maybe? :)

Reply:
Do you mean use "2" (which actually needs to be 3 including null-term) instead of "TOTP_str_len - curr_index" as third argument?

Also, will make "2" a #define since it is used 3 places...

Reply:
oops.... 3 indeed :)